src/coreclr/nativeaot/docs/troubleshooting.md

```diff
@@ -7,3 +7,4 @@ Sometimes you want to have more information how NativeAOT works. The compiler pr
 * `<IlcGenerateMapFile>true</IlcGenerateMapFile>`: Generates log files `ProjectName.map.xml` which describe layout of objects.
 * `<IlcSingleThreaded>true</IlcSingleThreaded>`: Perform compilation on single thread.
 * `<IlcDumpGeneratedIL>true</IlcDumpGeneratedIL>`: Dump IL for method bodies the compiler generated on the fly into `ProjectName.il`. This can be helpful when debugging IL generation - e.g. marshalling. The compiler maps debug information to this file, so it's possible to step in it and set breakpoints.
+* `<IlcGenerateMstatFile>true</IlcGenerateMstatFile>`: Generation of mstat file which can be parsed using. The mstat file is a binary file which can be read by a tool like: https://gist.github.com/MichalStrehovsky/2c7cb3d623c7f8901541914dab04238d. Just change the file path on line 12 to your app's.
```

Comment:
I'd like to sort out the attribution for that gist if we're to put it in the doc - the code is from @Suchiman - https://github.com/dotnet/runtime/pull/73913#issuecomment-1214374600. It needed a tweak after doing a breaking change to the format in https://github.com/dotnet/runtime/pull/75328 so I put it in a gist.

@Suchiman do you have a preference for how to handle it? Can we update your original comment with the file format change and link to that?

Reply:
I'm pretty flexible in that regards, whatever works best for you, however i cannot edit the PR post anymore as that has been `locked as resolved and limited conversation to collaborators` which prevents edits to existing comments as well it seems

Reply:
I can still edit it - I've made the change. We should instead link to https://github.com/dotnet/runtime/pull/73913#issuecomment-1214374600 so that the authorship is clear.

Reply:
@Suchiman I can send you a lot hugs for that change. Can you take a look, if current version is clear enough?